sympy/stats/crv_types.py

```diff
@@ -2276,6 +2278,84 @@ def StudentT(name, nu):
     return rv(name, StudentTDistribution, (nu, ))
 
 #-------------------------------------------------------------------------------
+# Trapezoidal distribution ------------------------------------------------------
+
+class TrapezoidalDistribution(SingleContinuousDistribution):
+    _argnames = ('a', 'b', 'c', 'd')
+
+    def pdf(self, x):
+        a, b, c, d = self.a, self.b, self.c, self.d
+        return Piecewise(
+            (2*(x-a) / ((b-a)*(d+c-a-b)), And(a <= x, x < b)),
+            (2 / (d+c-a-b), And(b <= x, x < c)),
+            (2*(d-x) / ((d-c)*(d+c-a-b)), And(c <= x, x <= d)),
+            (S.Zero, True))
+
+def Trapezoidal(name, a, b, c, d):
+    r"""
+    Create a continuous random variable with a trapezoidal distribution.
+
+    The density of the trapezoidal distribution is given by
+
+    .. math::
+        f(x) := \begin{cases}
+                  0 & \mathrm{for\ } x < a, \\
+                  \frac{2(x-a)}{(b-a)(d+c-a-b)} & \mathrm{for\ } a \le x < b, \\
+                  \frac{2}{d+c-a-b} & \mathrm{for\ } b \le x < c, \\
+                  \frac{2(d-x)}{(d-c)(d+c-a-b)} & \mathrm{for\ } c \le x < d, \\
+                  0 & \mathrm{for\ } d < x.
+                \end{cases}
+
+    Parameters
+    ==========
+
+    a : Real number, :math:`a \in \left(-\infty, \infty\right)`
```

Comment:
I would prefer writing them as written on wikipedia article i.e.

  | a                  (         a         <         d         )                 {\displaystyle a\;(a<d)}     - lower bound                         b                  (         a         ≤         b         <         c         )                 {\displaystyle b\;(a\leq b<c)}     - level start                         c                  (         b         <         c         ≤         d         )                 {\displaystyle c\;(b<c\leq d)}     - level end                         d                  (         c         ≤         d         )                 {\displaystyle d\;(c\leq d)}     - upper bound
-- | --

Reply:
Only this comment needs to be addressed. Apart from that this PR look good.